server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -147,6 +147,16 @@ public final class IndicesRequestCache implements RemovalListener<ICacheKey<Indi
         Property.NodeScope
     );
 
+    /**
+     * If enabled, allows caching size > 0 queries.
+     */
+    public static final Setting<Boolean> ALLOW_SIZE_NONZERO_SETTING = Setting.boolSetting(
+        "indices.requests.cache.allow_size_nonzero",
```

Comment:
I would avoid naming it like this. 
Better to simply use `indices.requests.cache.enable` ? So that later we can generalize for any other condition as well rather than only non_zero.
Also update the java doc to say that this setting allows us to make all the requests eligible for caching in request cache or something similar.

Reply:
I don't think we should call it `indices.requests.cache.enable`, since that'd be confusing with the existing setting to entirely enable/disable the request cache. It doesn't quite make all requests eligible for caching either, because we still can't cache things like DFS queries, queries with the `now` keyword, etc.

I think it'd be better to keep the naming specific to size > 0 since that's what it does right now, and if we later add the more generic system where the decision is delegated to the cache implementations, the relevant settings would be attached to those cache implementations like `indices.requests.cache.tiered_spillover.[setting]`?

Reply:
I don't think we have any setting which enables/disabled the setting as a whole, right? We just have index level setting.
Maybe we can choose a separate naming but I don't want us to name it specific to `allow_non_zero` as it is one of the condition we check to cache items. We can maybe add another Y policy/condition for request cache in the future, I don't think we will create another setting to enable/disable that?

Reply:
>It doesn't quite make all requests eligible for caching either, because we still can't cache things like DFS queries, queries with the now keyword, etc.

This is fine. We can call it out explicitly in the documentation. Anyway even today users can pass `?request_cache=true` but it won't necessarily cache it in request cache. So the behaviour is the same.

Reply:
How about `enable_for_all_requests`? This makes it clearer that it's about expanding the category of requests that can enter the cache, without implying the request cache is disabled when it's set to false. Then in the javadoc + doc page we can clarify that DFS queries, etc, are still not cacheable, and it's only applying to the queries that *can* be cached.